### DISCUSSION TITLE
Setting entry points (e.g. gates or current position) for a field when planning route

### DISCUSSION THREAD

**Discussion by tschuette22** [Fields2Cover/Fields2Cover#180] [Q&A]
Hi everyone,
first of all thank you for your work on this package.
I‘m curious if it‘s possible to set entry positions when planning the route on existing swaths.
I have not found anything on this in the documentation yet, so I‘m curious how you would do this.

The idea is to be able to plan the route starting from a given point on the field or with a given swath and direction. In our application the idea would be to first do the swath that‘s closest to the current robot position.

**Comment by Gonzalo-Mier** [comment]
Hi @tschuette22,

Nowadays we support setting the start and end points as the same thing. PR #177 supports different start and end points for the route.

If you want to start with the closest swath, I would suggest to remove that swath from your collection of swaths, and set the start point at the end of that swath. Then in the route, you start adding your swath first, and then adding the generated route.

**Reply by tschuette22** [reply]
Thanks a lot. That sounds like it provides what I want to do. I will look into it.
